sklearn/manifold/t_sne.py

```diff
@@ -650,10 +650,11 @@ def _fit(self, X, skip_num_points=0):
         ----------
         X : array, shape (n_samples, n_features) or (n_samples, n_samples)
             If the metric is 'precomputed' X must be a square distance
-            matrix. Otherwise it contains a sample per row. Note that this
-            when method='barnes_hut', X cannot be a sparse array and if need be
-            will be converted to a 32 bit float array. Method='exact' allows
-            sparse arrays and 64bit floating point inputs.
+            matrix. Otherwise it contains a sample per row. Note that
+            when method='barnes_hut', X cannot be a sparse array and
+            will be converted to a 32 bit float array if need be.
```

Comment:
```suggestion
            will be converted to a 32 bit float array if needed.
```

Reply:
I'm fine with "if need be", it's proper English